ext/loggly/loggly.go

```diff
@@ -0,0 +1,52 @@
+package loggly
+
+import (
+	"fmt"
+
+	"github.com/segmentio/go-loggly"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+// LogglyHook implements log15.Handler. A LogglyHook must be created by NewLogglyHook.
+// Exported fields can be modified during setup, but should not be touched.
+type LogglyHook struct {
+	// Client to the loggly api's
+	Client *loggly.Client
+}
+
+// NewLogglyHook creates
+func NewLogglyHook(token string) *LogglyHook {
+	return &LogglyHook{
+		Client: loggly.New(token),
+	}
+}
+
+// Log implements log15.Handler and sends the provided record to loggly
+func (lh *LogglyHook) Log(r *log15.Record) error {
+	data := make(map[string]interface{}, len(r.Ctx)/2)
+	for i := 0; i < len(r.Ctx); i += 2 {
+		key := r.Ctx[i]
+		value := r.Ctx[i+1]
+		keyStr, ok := key.(string)
+		if !ok {
+			keyStr = fmt.Sprintf("%v", key)
+		}
+		data[keyStr] = value
+	}
+	msg := loggly.Message{
+		"message":   r.Msg,
+		"level":     r.Lvl.String(),
+		"timestamp": r.Time,
+		"data":      data,
```

Comment:
Loggly provides excellent searching on nested data. I wanted to add level, message and timestamp because that is expected behavior, but I didn't want it to conflict with any data the user might have set.